src/rpcdataacq.cpp

```diff
@@ -0,0 +1,730 @@
+
+        if(i_research>0)
+        {
+            const double i_magnitude = cur->nMagnitude;
+            sum_magnitude= sum_magnitude + i_magnitude;
+            cnt_research += 1;
+        }
+
+        blockcount++;
+        samples++;
+        if(samples>=smoothing)
+        {
+            int midheight = cur->nHeight + (smoothing/2);
+            double samples_w_cpid = samples - cnt_investor;
+            if(samples == cnt_investor)
+                samples_w_cpid = std::numeric_limits<double>::max();
+            double samples_w_research = cnt_research;
+            if(cnt_research==0)
+                samples_w_research = std::numeric_limits<double>::max();
+            double samples_w_trans = (samples-cnt_empty);
+            if(samples==cnt_empty)
+                samples_w_trans = std::numeric_limits<double>::max();
+            points++;
+            Output << midheight << "  ";
+
+            Output << (sum_diff / samples) << " " << min_diff << " " << max_diff << "  ";
+            Output << (sum_spacing / samples) << " " << min_spacing << " " << max_spacing << "  ";
+            Output << (sum_size / samples) << " " << min_size << " " << max_size << "  ";
+            Output << (sum_research / samples) << " " << (sum_research / samples_w_research) << " " << max_research << "  ";
+            Output << (sum_interest / samples) << " " << max_interest << "  ";
+            Output << (cnt_empty / samples) << " " << cnt_empty << "  ";
+            Output << (cnt_investor / samples) << " " << cnt_investor << "  ";
+            Output << (cnt_trans / samples) << " " << (cnt_trans / samples_w_trans) << " " << cnt_trans << "  ";
+            Output << (cnt_research / samples) << " " << cnt_research << "  ";
+            Output << (cnt_contract / samples) << " " << cnt_contract << "  ";
+            Output << (cnt_neuralvote / samples) << " " << cnt_neuralvote << "  ";
+            Output << (cnt_neuralcurr / samples) << " " << cnt_neuralcurr << "  ";
+            Output << (sum_magnitude / samples_w_cpid) << "  ";
+            // missing: trans, empty, size, neural
+
+            Output << "\n";
+            samples = 0;
+            sum_diff = 0;
+            min_diff = INT_MAX;
+            max_diff = 0;
+            cnt_empty = 0;
+            cnt_investor = 0;
+            sum_spacing = 0;
+            min_spacing = INT_MAX;
+            max_spacing = 0;
+            sum_size = 0;
+            min_size = INT_MAX;
+            max_size = 0;
+            cnt_trans = 0;
+            sum_research = 0;
+            max_research = 0;
+            sum_interest = 0;
+            max_interest = 0;
+            sum_magnitude = 0;
+            cnt_research = 0;
+            cnt_neuralvote = 0;
+            cnt_neuralcurr = 0;
+            cnt_contract = 0;
+        }
+        /* This is wery important */
+        cur = cur->pprev;
+    }
+
+    result1.push_back(Pair("file", o_path.string()));
+    result1.push_back(Pair("points",points));
```

Comment:
This needs to be explicit for some compilers. It works for example if you cast with `(boost::uint64_t)points`

Reply:
Just `uint64_t` should be fine as we have it in the compiler. If we switch the variable `uint64_t` it should work on all platforms. If not, an `uint32_t` should be sufficient.

Reply:
Double? :laughing: